extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/dialect/QuarkusH2Dialect.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.hibernate.orm.deployment.dialect;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+
+import org.hibernate.dialect.H2Dialect;
+import org.hibernate.engine.jdbc.env.spi.IdentifierCaseStrategy;
+import org.hibernate.engine.jdbc.env.spi.IdentifierHelper;
+import org.hibernate.engine.jdbc.env.spi.IdentifierHelperBuilder;
+
+/**
+ * Subclass of H2Dialect fixing schema updates by considering unquoted identifiers as upper case.
+ * This is H2's default.
+ * See https://github.com/quarkusio/quarkus/issues/1886
+ *
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public class QuarkusH2Dialect extends H2Dialect {
```

Comment:
The 2 dialect classes need to be in the `runtime` artifact.

Otherwise it won't work very well at runtime :).